libtiledbsoma/src/soma/soma_binary_column.cc

```diff
@@ -0,0 +1,404 @@
+            reinterpret_cast<const char*>(bytes.data()), bytes.size()));
+    }
+
+    query.select_points(name(), casted_points);
+}
+
+void SOMABinaryColumn::_set_dim_ranges(
+    ManagedQuery& query, const std::any& ranges) const {
+    if (!isIndexColumn()) {
+        throw TileDBSOMAError(fmt::format(
+            "[SOMABinaryColumn][_set_dim_ranges] Column with name {} is not an "
+            "index "
+            "column",
+            name()));
+    }
+
+    std::vector<std::pair<std::string, std::string>> casted_ranges;
+
+    for (const auto& bytes : std::any_cast<std::vector<
+             std::pair<std::vector<std::byte>, std::vector<std::byte>>>>(
+             ranges)) {
+        casted_ranges.push_back(std::make_pair(
+            std::string(
+                reinterpret_cast<const char*>(bytes.first.data()),
+                bytes.first.size()),
+            std::string(
+                reinterpret_cast<const char*>(bytes.second.data()),
+                bytes.second.size())));
+    }
+
+    query.select_ranges(name(), casted_ranges);
+}
+
+void SOMABinaryColumn::_set_current_domain_slot(
+    NDRectangle& rectangle, std::span<const std::any> domain) const {
+    if (!isIndexColumn()) {
+        throw TileDBSOMAError(fmt::format(
+            "[SOMABinaryColumn][_set_current_domain_slot] Column with name {} "
+            "is "
+            "not "
+            "an index column",
+            name()));
+    }
+
+    if (domain.size() != 1) {
+        throw TileDBSOMAError(fmt::format(
+            "[SOMABinaryColumn][_set_current_domain_slot] Invalid domain size. "
+            "Expected 1, got {}",
+            domain.size()));
+    }
+
+    auto dom = std::any_cast<std::array<std::vector<std::byte>, 2>>(domain[0]);
+    if (dom[0].size() == 0 && dom[1].size() == 0) {
+        rectangle.set_range(name(), "", "\x7f");
+    } else {
+        throw TileDBSOMAError(fmt::format(
+            "[SOMABinaryColumn][_set_current_domain_slot] domain (\"{}\", "
+            "\"{}\") cannot be set for "
+            "binary index columns: please use "
+            "(\"\", \"\")",
+            std::string(
+                reinterpret_cast<const char*>(dom[0].data()), dom[0].size()),
+            std::string(
+                reinterpret_cast<const char*>(dom[1].data()), dom[1].size())));
+    }
+}
+
+std::pair<bool, std::string> SOMABinaryColumn::_can_set_current_domain_slot(
+    std::optional<NDRectangle>&, std::span<const std::any> new_domain) const {
+    if (!isIndexColumn()) {
+        throw TileDBSOMAError(fmt::format(
+            "[SOMABinaryColumn][_set_current_domain_slot] Column with name {} "
+            "is "
+            "not "
+            "an index column",
+            name()));
+    }
+
+    if (new_domain.size() != 1) {
+        throw TileDBSOMAError(fmt::format(
+            "[SOMABinaryColumn][_can_set_current_domain_slot] Expected domain "
+            "size for '{}' is 1, found {}",
```

Comment:
I won't keep adding comments for every one of these -- please re-join the over-split string-literal lines throughout this file.